bin/pre-commit.sh

```diff
@@ -1,46 +1,25 @@
 #!/bin/bash
-# Runs pre-commit tests.
-#
-# Instead of failing on first error, complete all checks, then fail if need be.
 
-# Echo go's version to verify against go's behavior below.
-# Different versions do different things with respect to vendored code.
-go version
+rc=0
 
-# Assert state.
-if [ -n "$failIt" ]; then
-  echo "Expecting failIt to be empty."
-  exit 1
-fi
+go_dirs() {
+  go list -f '{{.Dir}}' ./... | tr '\n' '\0'
```

Comment:
I like that. The only thing which is not supported now are paths with `\n`, of course. But I think it is fair enough to not support that (at least for now).

Reply:
Yeah, that'd be pretty unfortunate.